src/WidgetBase.ts

```diff
@@ -160,6 +181,62 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 				propertiesChangedListeners[i].call(this, evt);
 			}
 		}));
+
+		// Add a render decorator that will register callbacks for the v-node 'afterCreate' and
+		// 'afterUpdate' lifecycle methods to call the widget lifecycle methods onElementCreated
+		// and onElementUpdated
+		this.addDecorator('afterRender', (node: DNode) => {
+			// Cannot use @dojo/core/aspect because the vnode callback functions cannot change on each
+			// call to render.
+			decorate(node, (node: HNode) => {
+				const key = node.properties && node.properties.key;
+				if (key != null) {
+					this.setLifecycleProperties(node.properties, 'afterCreate');
+					node.properties.afterCreate = vnodeAfterCreate;
+				}
+			}, isHNode);
+
+			decorate(node, (node: HNode) => {
+				const key = node.properties && node.properties.key;
+				if (key != null) {
+					this.setLifecycleProperties(node.properties, 'afterUpdate');
+					node.properties.afterUpdate = vnodeAfterUpdate;
+				}
+			}, isHNode);
+			return node;
+		});
+	}
+
+	private setLifecycleProperties(properties: VirtualDomProperties, callbackKey: string) {
+		let lsProps = properties._lifecycleProperties;
+		if (!lsProps) {
+			properties._lifecycleProperties = lsProps = {
+				widget: this
+			};
+		}
+		if (!lsProps.hasOwnProperty(callbackKey)) {
+			lsProps[callbackKey] = properties[callbackKey];
+		}
+	}
+
+	/**
+	 * Widget lifecycle method that is called whenever a vdom node is created.
+	 * Override this method to access the dom nodes that were inserted into the dom.
+	 * @param element The dom node represented by the vdom node.
+	 * @param key The vdom node's key.
+	 */
+	protected onElementCreated(element: Element, key: string): void {
+		// Do nothing by default.
+	}
+
+	/**
+	 * Widget lifecycle method that is called whenever a vdom node is updated.
+	 * Override this method to access the dom nodes that were updated in the dom.
+	 * @param element The dom node represented by the vdom node.
+	 * @param key The vdom node's key.
+	 */
+	protected onElementUpdated(element: Element, key: string): void {
```

Comment:
I'm starting to be skeptical on how useful maquette's `afterUpdate` function actually would be to the end user. It seems like its called regardless of whether the specific dom node is updated or not, which kind of makes me think if we decide to keep this lifecycle hook we should perhaps only call it once per update (and not per node).